File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesAdapter.java
package paul.barthuel.go4lunch.ui.workmates;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import paul.barthuel.go4lunch.R;

public class WorkmatesAdapter extends ListAdapter<WorkmatesInfo, WorkmatesAdapter.ViewHolder> {

    private final Listener listener;

    public WorkmatesAdapter(Listener listener) {
        super(new WorkmatesInfoDiffCallBack());
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.workmates_item, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int index) {
        viewHolder.bind(getItem(index), listener);
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView mTextViewName;
        private final TextView mTextViewRestaurantName;
        private final ImageView mImageViewThumbnail;

        ViewHolder(@NonNull View itemView) {
            super(itemView);

            mTextViewName = itemView.findViewById(R.id.workmates_item_tv_username);
            mImageViewThumbnail = itemView.findViewById(R.id.workmates_item_iv_thumbnail);
            View mRoot = itemView.findViewById(R.id.workmates_item_cl);
            mTextViewRestaurantName = itemView.findViewById(R.id.workmates_item_tv_restaurant_name);
        }

        private void bind(final WorkmatesInfo workmatesInfo,
                          final Listener listener) {

            mTextViewName.setText(workmatesInfo.getName());
            Glide.with(mImageViewThumbnail)
                    .load(workmatesInfo.getImage())
                    .apply(RequestOptions.circleCropTransform())
                    .into(mImageViewThumbnail);
            mImageViewThumbnail.setOnClickListener(v -> listener.onWorkmateInfoClick(workmatesInfo));
            mTextViewRestaurantName.setText(workmatesInfo.getRestaurantName());
            mTextViewRestaurantName.setOnClickListener(v -> listener.onRestaurantClick(workmatesInfo.getPlaceId(), workmatesInfo.getRestaurantName()));
        }
    }

    public interface Listener {
        void onWorkmateInfoClick(WorkmatesInfo workmatesInfo);
        void onRestaurantClick(String id, String restaurantName);
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesAdapter.kt
package paul.barthuel.go4lunch.ui.workmates

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import paul.barthuel.go4lunch.R

class WorkmatesAdapter(private val listener: Listener) : ListAdapter<WorkmatesInfo, WorkmatesAdapter.ViewHolder>(WorkmatesInfoDiffCallBack()) {
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.workmates_item, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, index: Int) {
        viewHolder.bind(getItem(index), listener)
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val mTextViewName = itemView.findViewById<TextView>(R.id.workmates_item_tv_username)
        private val mTextViewRestaurantName = itemView.findViewById<TextView>(R.id.workmates_item_tv_restaurant_name)
        private val mImageViewThumbnail = itemView.findViewById<ImageView>(R.id.workmates_item_iv_thumbnail)
        fun bind(workmatesInfo: WorkmatesInfo?,
                         listener: Listener) {
            mTextViewName.text = workmatesInfo!!.name
            Glide.with(mImageViewThumbnail)
                    .load(workmatesInfo.image)
                    .apply(RequestOptions.circleCropTransform())
                    .into(mImageViewThumbnail)
            mImageViewThumbnail.setOnClickListener { listener.onWorkmateInfoClick(workmatesInfo) }
            mTextViewRestaurantName.text = workmatesInfo.restaurantName
            mTextViewRestaurantName.setOnClickListener { listener.onRestaurantClick(workmatesInfo.placeId, workmatesInfo.restaurantName) }
        }

    }

    interface Listener {
        fun onWorkmateInfoClick(workmatesInfo: WorkmatesInfo?)
        fun onRestaurantClick(id: String?, restaurantName: String?)
    }

}